app/src/marketplace/containers/EditProductPage2/EditControllerProvider.jsx

```diff
@@ -30,111 +24,70 @@ const EditControllerContext: Context<ContextProps> = React.createContext({})
-                // save product
-                await putProduct(product, product.id || '')
-
-                // upload image
-                if (product.newImageToUpload != null) {
-                    try {
-                        await postImage(product.id || '', product.newImageToUpload)
-                    } catch (e) {
-                        console.error('Could not upload image', e)
-                    }
-                }
+    const { api: deployCommunityDialog } = useModal('deployCommunity')
+    const { api: confirmSaveDialog } = useModal('confirmSave')
 
-                // TODO: check errors
-                return true
-            })
-
-            // Check update for contract product
-            if (savedSuccessfully) {
-                savedSuccessfully = await contractSavePending.wrap(async () => {
-                    // fetch contract product from public node to see if we need to update the contract product
-                    let contractProduct
-
-                    try {
-                        contractProduct = await getProductFromContract(product.id || '', true)
-                    } catch (e) {
-                        console.warn(e)
-                    }
-
-                    // use original product to check if it was paid or not
-                    if (!!contractProduct && isPaidProduct(originalProduct) && isUpdateContractProductRequired(contractProduct, product)) {
-                        // start contract transaction dialog, this will take care of checking web3
-                        // and fetching/updating the contract product
-                        const contractUpdated = await updateContractDialog.open({
-                            product,
-                            originalProduct,
-                            contractProduct,
-                        })
-
-                        return contractUpdated
-                    }
-
-                    return true
-                })
-            }
+    const productId = product.id
+    const redirectToProduct = useCallback(() => {
+        if (!isMounted()) { return }
+        history.replace(routes.product({
+            id: productId,
+        }))
+    }, [
+        productId,
+        isMounted,
+        history,
+    ])
 
-            // Everything ok, do a redirect back to product page
-            if (savedSuccessfully) {
-                if (!isMounted()) { return }
-                history.replace(formatPath(links.marketplace.products, product.id))
+    const save = useCallback(async () => {
+        const savedSuccessfully = await savePending.wrap(async () => {
+            // save product
+            await putProduct(product, product.id || '')
+
+            // upload image
+            if (product.newImageToUpload != null) {
+                try {
+                    await postImage(product.id || '', product.newImageToUpload)
+                } catch (e) {
+                    console.error('Could not upload image', e)
```

Comment:
Yep, not sure should we break the flow here - would be good to inform the user that it failed. Anyway, this will change a bit so that the url is stored into intermediate storage if editing a published product.